### DISCUSSION TITLE
fonoster for dtmf, bidirectional audio stream and asterisk

### DISCUSSION THREAD

**Discussion by CHETAN-SEHGAL** [fonoster/yaps#645] [General]
there ar two npm fonoster packages availabe, 
1) Fonoster/Stream Package:
Provides raw audio bidirectional streaming
Supports direct audio buffer playback using res.write(data)
I'll need to verify its DTMF detection capabilities. ( i still have not found if this package provide dtmf detection, if this package provide that please do let me know how to implement the same)

2)Fonoster/Voice Package:
Confirmed to provide DTMF detection functionality
I'll investigate if it supports direct audio stream playback like res.stream(data). ( still have not found if this, if this package provide that please do let me know how to implement the same)

or can i use both the package at same time in a node file, if it does not raise a conflict. any help regarding this gonna be appreciated. thanks .

**Comment by psanders** [comment]
Hi @CHETAN-SEHGAL

The `@fonoster/stream` module is limited by what's possible with Asterisk Audiosocket. 

As for the `@fonoster/voice,` streaming media has not been fully implemented.

Are you planning to use `@fonoster/voice` in combination with Fonoster or standalone?
